weblab/datasets/views.py

```diff
@@ -230,3 +235,37 @@ def test_func(self):
     def get_success_url(self, *args, **kwargs):
         ns = self.request.resolver_match.namespace
         return reverse(ns + ':list')
+
+
+class DatasetRenameView(LoginRequiredMixin, UserFormKwargsMixin, UserPassesTestMixin, FormMixin, DetailView):
+    template_name = 'datasets/dataset_rename_form.html'
+    context_object_name = 'dataset'
+    """
+    Delete dataset
+    """
+    model = Dataset
+    form_class = DatasetRenameForm
+
+    def _get_object(self):
+        if not hasattr(self, 'object'):
+            self.object = self.get_object()
+        return self.object
+
+    def test_func(self):
+        return self._get_object().is_managed_by(self.request.user)
+
+    def post(self, request, *args, **kwargs):
+        form = self.get_form()
+
+        if form.is_valid():
+            new_name = form.cleaned_data['name']
+            dataset = self.get_object()
+            dataset.name = new_name
```

Comment:
You'll probably need to implement the archive name here. Something like:
```suggestion
            old_archive_path = dataset.archive_path
            dataset.name = new_name
            if old_archive_path.exists():
                old_archive_path.rename(dataset.archive_path)
```